### PR TITLE
Add an upgrade guide

### LINKED ISSUE #274: Add upgrade doc
see 

- https://github.com/spiral-project/ihatemoney/issues/264#issuecomment-327219245
- https://github.com/spiral-project/ihatemoney/issues/264#issuecomment-338497339

### PR DESCRIPTION
**Pull request by JocelynDelalande**
fix #274 

We were at risk of having an empty PR queue, so here is my attempt to fill it a bit ;-).

I'm hungry for comments/feedback, as I'm attaching importance to documentation :).

I think after this PR and #277 are last blockers for #264.

### CODE REVIEW

**Comment by JocelynDelalande** [comment]
@0livd specially interested in your feedback, as you are the official apache guru here !

PS: « Apache guru » sounds realy great ;-)

File: docs/upgrade.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+2. Install ihatemoney with pip::
+
+    pip install ihatemoney
+
+3. Fix your configuration file (paths *have* changed), depending on the software you use in your setup:
+
+   - **gunicorn**: ``ihatemoney generate-config gunicorn.conf.py`` (nothing
+     critical changed, keeping your old config might be fine)
+
+   - **supervisor** : ``ihatemoney generate-config gunicorn.conf.py`` (mind the
```
Comment (0livd) [inline docs/upgrade.rst]: I guess it is supervisord.conf instead of gunicorn.conf.py ?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -31,6 +31,7 @@ Added
 - ACTIVATE_ADMIN_DASHBOARD setting (#262)
 - Link to the dashboard in the navigation bar (#262)
 - Dockerfile
+- Documentation explaining upgrade process
```
Comment (almet) [inline CHANGELOG.rst]: nit: explaining *the* upgrade process.

File: docs/upgrade.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+When upgrading from a major version to another, you **must** follow special
+instructions:
+
+1.x → 2.x
+---------
+
+Switch from git installation to pip installation
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+The recomended installation method is now using pip (git is now intended for
```
Comment (almet) [inline docs/upgrade.rst]: typo: *recommended*

Comment (almet) [reply docs/upgrade.rst]: I would remove the parenthesis for git, and put it as a separate sentence.

File: docs/upgrade.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+development only).
+
+.. warning:: Be sure not to delete by accident your sqlite db or your settings
```
Comment (almet) [inline docs/upgrade.rst]: nit: rephrase: "Be extra careful to not remove your sqlite database nor your settings file, if they are stored inside the cloned folder".

File: docs/upgrade.rst [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+             file if you stored it within the cloned folder.
+
+1. Delete the cloned folder
+
+
+.. note:: the following are to be run from your virtualenv, if any, (see
```
Comment (almet) [inline docs/upgrade.rst]: nit: I would rephrase: "If you are using a virtualenv, then the following commands should be run inside it".

**Comment by JocelynDelalande** [comment]
Thanks for the comments @0livd   @almet I adressed your comments and merged :-)

**Comment by almet** [comment]
great, thanks!
